apps/editing-toolkit/editing-toolkit-plugin/tutorials/class-wpcom-tutorials.php

```diff
@@ -0,0 +1,246 @@
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Registers a Tutorial.
+	 *
+	 * @param string $id      A unique ID for the Tutorial.
+	 * @param array  $options `title` and `tasks` required.
+	 *
+	 * @return bool           True if successfully registered, false if it failed.
+	 */
+	public function register( $id, $options ) {
+		if ( ! $this->validate_options( $options ) ) {
+			return false;
+		}
+		$options['id']         = $id;
+		$this->registry[ $id ] = $options;
+		return true;
+	}
+
+	/**
+	 * Unregisters a registered Tutorial
+	 *
+	 * @param string $id Tutorial ID.
+	 * @return bool      True if successfully unregistered, false if not found.
+	 */
+	public function unregister( $id ) {
+		if ( ! isset( $this->registry[ $id ] ) ) {
+			return false;
+		}
+		unset( $this->registry[ $id ] );
+		return true;
+	}
+
+	/**
+	 * Validates the options passed in `WPCom_Tutorials::register`.
+	 *
+	 * @param array $options Associative array passed with registration.
+	 * @return bool          Validation success.
+	 */
+	private function validate_options( $options ) {
+		if ( ! isset( $options['title'] ) || ! is_string( $options['title'] ) ) {
+			return false;
+		}
+		if ( ! isset( $options['tasks'] ) || ! is_array( $options['tasks'] ) ) {
+			return false;
+		}
+		foreach ( $options['tasks'] as $task ) {
+			if ( ! $this->validate_task( $task ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Validates a single task for expected structure.
+	 *
+	 * @param array $task A single task (id and title).
+	 * @return bool       Validation success.
+	 */
+	private function validate_task( $task ) {
+		return isset( $task['id'] ) && isset( $task['title'] );
+	}
+
+	/**
+	 * Retrieves a Tutorial from the registry.
+	 *
+	 * @param string $id  The ID of the Tutorial to fetch.
+	 * @return null|array Tutorial associative array if found, null if not.
+	 */
+	public function get_tutorial( $id ) {
+		if ( ! isset( $this->registry[ $id ] ) ) {
+			return null;
+		}
+		$tutorial = $this->registry[ $id ];
+
+		// Populating status at runtime so that any invalidations happen within the Options API.
+		$tutorial['tasks'] = $this->get_tasks( $tutorial );
+		return $tutorial;
+	}
+
+	/**
+	 * Retrives all registered Tutorial IDs
+	 *
+	 * @return array list of registered Tutorial IDs.
+	 */
+	public function get_registered_ids() {
+		return array_keys( $this->registry );
+	}
+
+	/**
+	 * Gets registered tasks for a Tutorial, with stored completion status.
+	 *
+	 * @param array $tutorial Tutorial object.
+	 * @return array          Tasks for Tutorial, with task completion status.
+	 */
+	private function get_tasks( $tutorial ) {
+		$tasks_with_statuses = array();
+		$statuses            = $this->get_statuses( $tutorial['id'] );
+
+		foreach ( $tutorial['tasks'] as $task ) {
+			$task['status']        = isset( $statuses[ $task['id'] ] ) ? $statuses[ $task['id'] ] : 'pending';
```

Comment:
Could we assign a default 'pending' status to each task instead of dynamically assigning it in this ternary? In my opinion, it will make things more explicit.

Reply:
I will agree to 99.5% of ternary removal requests and this is no exception.